Javascript_2/Flight booking fullname function/script.js

```diff
@@ -0,0 +1,33 @@
+function getFullName(
+  firstName,
+  surName,
+  useFormalName = false,
+  isFemale = false
+) {
+  firstName = firstName.trim();
+  surName = surName.trim();
+
+  if (firstName === "" && surName === "") {
+    return "";
+  } else if (firstName === "") {
+    return surName;
+  } else if (surName === "") {
+    return firstName;
+  }
+  let fullName = firstName + " " + surName;
+
+  if (useFormalName) {
+    if (isFemale) {
+      fullName = "Lady " + fullName;
+    } else {
+      fullName = "Lord " + fullName;
+    }
+  }
+  return fullName;
+}
+
+let fullName1 = getFullName("Benjamin", "Hughes", true);
+let fullName2 = getFullName("Julia", "Anderson", true, true);
```

Comment:
Imagine you've created the function and logic and you get the data from another place and you can't relly for it to be always correct. Imagine the function is called in the following way `let fullName2 = getFullName("Julia", "Anderson", true);` what would happen then is that you will print "Lord Julia Anderson". Would be nice to skip the formal greeting if it's `true` when there isn't a 4rth argument passed to avoid this edge case.
You are welcome to implement it, but otherwise the solution is working very well.

Reply:
I tried but i can't fix it. I have no idea :(

Reply:
Here it is:
```
  if (useFormalName && (isFemale !== undefined ) ) {
    if (isFemale) {
      fullName = "Lady " + fullName;
    } else {
      fullName = "Lord " + fullName;
    }
  }
  return fullName;
  ```
  This will cover all possible scenarios, and especially this: 
  ```
  const fullName2 = getFullName("Julia", "Anderson", true);
  ```
  
  The `isFemale !== undefined` check can be replaced by `isFemale === true || isFemale === false `, but its longer.